sw/device/lib/crypto/impl/hash.c

```diff
@@ -0,0 +1,96 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/include/hash.h"
+
+#include <stdbool.h>
+
+#include "sw/device/lib/crypto/drivers/kmac.h"
+
+/**
+ * Return the digest size (in bytes) for given hashing mode.
+ *
+ * @param hash_mode Hashing mode (e.g. kHashModeSha256).
+ * @param digest_len Result digest size in bytes.
+ * @return Error status.
+ */
+OT_WARN_UNUSED_RESULT
+static crypto_status_t get_digest_size(hash_mode_t hash_mode,
+                                       size_t *digest_len) {
+  if (digest_len == NULL) {
+    return kCryptoStatusBadArgs;
+  }
+
+  // Below `digest_len` is in bytes, therefore magic values are obtained
+  // after division by 8.
+  switch (hash_mode) {
+    case kHashModeSha3_224:
+      *digest_len = 224 / 8;
+      break;
+    case kHashModeSha256:
+    case kHashModeSha3_256:
+      *digest_len = 256 / 8;
+      break;
+    case kHashModeSha384:
+    case kHashModeSha3_384:
+      *digest_len = 384 / 8;
+      break;
+    case kHashModeSha512:
+    case kHashModeSha3_512:
+      *digest_len = 512 / 8;
+      break;
+    default:
+      return kCryptoStatusBadArgs;
+  }
+  return kCryptoStatusOK;
+}
+
+OT_WARN_UNUSED_RESULT
+crypto_status_t otcrypto_hash(crypto_const_uint8_buf_t input_message,
+                              hash_mode_t hash_mode,
```

Comment:
nit: I'd consider flattening this function to `otcrypto_hash_sha3_224(...)` to simplify its implementation.

Reply:
IIUC, this sounds like a question about the cryptolib API interface (keep 7 hash functions as separate or combine them into one). I think @jadephilipoom has more context about the API decisions.

Reply:
Fatih is right -- this is a good point but we'll need to address it in the formal API if it's going to change. I'll put it on the agenda, but I think we should not block this PR on it since that meeting won't convene until January.